### ISSUE TITLE
ruTorrent not tracking Finished time any longer

### ISSUE THREAD

**Issue by jakobbg** [Novik/ruTorrent#1879]
Updated with latest code from git, ruTorrent stopped tracking Finished time a few weeks ago. What can possibly have made this happen? Local problem or something related to codebase? Plugin seedingtime is enabled and loaded.

![image](https://user-images.githubusercontent.com/142172/56252537-87329980-60b8-11e9-8f23-f7a2c46eeeaa.png)

**Comment by xombiemp** [comment]
I'm not sure what would cause that, but I wanted to let you know that it's still working for me so it is probably a local problem.  I'm running the latest from git too.
<img width="222" alt="Screen Shot 2019-04-17 at 9 53 17 AM" src="https://user-images.githubusercontent.com/690710/56302431-f8913d00-60f6-11e9-8afb-c9018760d90d.png">

**Comment by jakobbg** [comment]
My box:

$ uname -a
FreeBSD core2.grimstveit.no 11.2-RELEASE-p9 FreeBSD 11.2-RELEASE-p9 #0 fa113e14a11(releng/11.2): Wed Feb 20 11:23:56 CET 2019     root@core2.grimstveit.no:/usr/obj/usr/src/sys/CORE2  amd64

[jakobbg@core2:~/.local/rutorrent] master(+9/-8)+* ± git pull
Already up to date.
[jakobbg@core2:~/.local/rutorrent] master(+9/-8)+* ± git status
On branch master
Your branch is up to date with 'origin/master'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

        modified:   conf/config.php
        modified:   plugins/rss/conf.php

Untracked files:
  (use "git add <file>..." to include in what will be committed)

        .htaccess

no changes added to commit (use "git add" and/or "git commit -a")

**Comment by jakobbg** [comment]
This problems persists, I have no idea what might be the reason for this:

![image](https://user-images.githubusercontent.com/142172/58005772-b76dbd80-7ae6-11e9-8f7c-c5ca7c1aa66a.png)

**Comment by xombiemp** [comment]
Strange enough, I've started to see this issue.  For me the last few torrents don't have a Added or Finished value.
<img width="347" alt="Screen Shot 2019-06-19 at 11 53 10 AM" src="https://user-images.githubusercontent.com/690710/59788475-0a26ca80-9289-11e9-9e50-dc20fda1afd6.png">

**Comment by jakobbg** [comment]
This is still ongoing for me. It doesn't seem to matter whether they are added manually or via RSS (like the three debian torrents in the attached image).

![image](https://user-images.githubusercontent.com/142172/62009157-59033200-b15c-11e9-9d72-fc5f7e5fb0c9.png)

**Comment by DFDC** [comment]
This is an issue with the "seedingtime" plugin. You won't be able to recover the data for torrents that are complete and already have it (without losing ratio by using fast resume).

Here's how you can fix it going forward:

1. Plugins tab
2. Right click "seedingtime" and disable.
3. Right click it again, hover over launch, and disable.
4. Right click it yet again, hover over launch, and enable.
5. Refresh. New torrents will have their finished time.

I've never been able to find logs or anything to explain the **WHY** but I figured I would post what little knowledge I have while searching for a resolution of mine own.

**Comment by jakobbg** [comment]
I did a "Force recheck" on one torrent for one reason or the other, and suddenly the correct time for Finished was populated (like, "3w 16h ago"). Like magic. There was much rejoicing! What config am I missing to have this enabled for all new torrents?

**Comment by DFDC** [comment]
No joy on forcing a recheck populating the missing data here. Annoying.

Since I haven't been able to track down individual plugin logs and nothing appears in my rutorrent logs for this issue I honestly don't know where it lies.

**Comment by jakobbg** [comment]
I had, myself, timestamp in the "added" column, which seems to be a prerequisite for the Finished column to populate after a "Force recheck". Fixed 100s of missing Finished columns, but have one row where both Finished and Added still is missing.

**Comment by aSkib** [comment]
Have the similar issue with the 'Added' column. (Finished column is populated though).

Have also mentioned that adding a torrent does not trigger the 'added' history event (history plugin) neither ratio rules are applied (extratio plugin)

environment:
rtorrent 0.9.7 - ruTorrent 3.9
lighttpd web server

**Comment by thirstyraven** [comment]
I've the exact same issue as well.

Finished column is blank for for all, which sometimes gets filled by force rechecking. Added column is blank for many, including newly added torrents.

<img width="913" alt="Screen Shot 2019-12-23 at 10 22 22 AM" src="https://user-images.githubusercontent.com/57574115/71374303-16108380-256f-11ea-9fdf-5a6d59a59979.png">

rTorrent: 0.9.8, ruTorrent: v3.9

**Comment by Beet4** [comment]
Hi!

Recently (started maybe a few weeks ago) my setup stopped tracking Finished status of torrents. Finished column (from plugin seedingtime) is empty for new torrents, history tab does not track finished status and therefore Pushbullet events for finished does not work. 

I tried to update to my latest docker revision today (ruTorrent 3.10) but the issue persists.

Is this issue still ongoing for some people? Have anyone found a solution? Strange that it would just start randomly. I've tried the mentioned workaround above with disable/enable the seedingtime plugin but it did not help.

**Comment by BRTPOB** [comment]
It's still ongoing for me, and the only real workaround I've found, which isn't super great, is to restart the container and any new torrents should show the finished time properly...until it breaks again.

**Comment by YaPaY** [comment]
the problem started a few day ago, seeding time plugin disable enable or change time format method did not help. 0.9.7 rtorrent 3.10 rutorrent on Raspberry Pi 4

switching another skin, restart the raspberry doesn't work, either.

**Comment by msw1970** [comment]
I'm experiencing the same issue... Added column populated but Finished column isn't. This started last week when I moved house and had to power down my torrent server. Issue has been present ever since. I have restarted the server a number of times without it resolving the problem. The only thing that seems to work is running a force recheck on the seeding entry. This consistently seems to populate the Finished column but doesn't add a Finshed event in the History plugin. I'm yet to know whether this will also affect processing of ratio rules etc.

Currently running rtorrent 0.9.6 and rutorrent 3.9

**Comment by phene** [comment]
Please re-open, this is still an ongoing bug.

**Comment by G1tOnW1th1t** [comment]
Agree, still an issue (torrent leterally just finished and now seeding but finished field empty)

**Comment by derailius** [comment]
i have the same problem

**Comment by ghoto** [comment]
Same issue. Using Crazy Max docker image.

**Comment by Jolly-Pirate** [comment]
I've had that on some torrents, my workaround is to stop and start them again, the finished time will show up.

**Comment by stickz** [comment]
Please open a new issue report and include specific details about your setup. It's possible this could be a race condition with a shell script, like we fixed with the `erasedata` plugin. But I can't speculate on that, until I've ruled out other potential causes.
